tests/test_execution_context.py

```
@@ -215,6 +215,44 @@ def get_current_session():
+            return [
+                SimpleNamespace(
+                    event_type="stream_interrupted",
+                    data={"type": "stream_interrupted"},
+                )
+            ]
+
+    async def _emit(event_type: str, data: dict[str, Any]) -> None:
+        events.append((event_type, data))
+
+    core = PenguinCore.__new__(PenguinCore)
+    setattr(core, "conversation_manager", _ConversationManager())
```

Ruff (0.15.12)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)